proposals/csharp-12.0/collection-expressions.md

```diff
@@ -116,28 +116,22 @@ The following implicit *collection expression conversions* exist from a collecti
 
   * For each *element* `Ei` there is an *implicit conversion* to `T`.
 
-* To an *[inline array type](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/inline-arrays.md)* with *element type* `T` where:
+* To a *struct* or *class type* that implements `System.Collections.IEnumerable` where:
 
-  * For each *element* `Ei` there is an *implicit conversion* to `T`.
-
-* To a *type* that implements `System.Collections.IEnumerable` where:
-
-  * The *type* contains an applicable instance constructor that can be invoked with no arguments or invoked with a single argument for the 0-th parameter where the parameter has type `System.Int32` and name `capacity`.
+  * The *type* contains an applicable instance constructor that can be invoked with no arguments.
   * For each *expression element* `Ei` there is an applicable instance or extension method `Add` for a single argument `Ei`.
   * For each *spread element* `Si` there is an applicable instance or extension method `Add` for a single argument of the *iteration type* of `Si`.
 
-  *Open issue: Relying on a parameter named `capacity` seems brittle. Is there an alternative?*
```

Comment:
i don't like that we got rid of capacity.  IMO this is an optimization that important for our perf story.

Reply:
We haven't removed the potential use of `.ctor(int capacity)`, it still might be used during construction, but the availability of that constructor shouldn't affect convertibility.

Reply:
When I parsed the new text, I interpreted it to mean the parameterless ctor is mandatory, and the `(int capacity)` ctor is optional. Meaning that if a type had a ctor with one parameter named `capacity`, but did not have an accessible parameterless ctor, that type couldn't participate in collection expressions. Is that the intent? 

I can see a case for both choices, so mostly just curious.

Reply:
Yes, a constructor that can be invoked with no arguments is required for the *collection expression conversion*, and the constructor with `int capacity` parameter was optional.